tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/Serverless.cs

```diff
@@ -49,47 +49,65 @@ internal static NativeCallTargetDefinition[] GetServerlessDefinitions()
+                var paramCount = handler.ParamTypeArray.Length;
+                var integrationType = handler.ParamTypeArray[0].StartsWith(ClrNames.Task) ? GetAsyncIntegrationTypeFromParamCount(paramCount) :
+                    GetSyncIntegrationTypeFromParamCount(paramCount);
                 return new NativeCallTargetDefinition[]
                 {
                     new(handler.GetAssembly(), handler.GetFullType(), handler.GetMethodName(), handler.ParamTypeArray, 0, 0, 0, 65535, 65535, 65535, assemblyName, integrationType)
                 };
             }
             catch (Exception ex)
             {
-                Serverless.Error("Impossible to get Serverless Definitions : " + ex.Message);
+                Error("Impossible to get Serverless Definitions", ex);
                 return Array.Empty<NativeCallTargetDefinition>();
             }
         }
 
-        internal static string GetIntegrationTypeFromParamCount(int paramCount)
+        internal static string GetAsyncIntegrationTypeFromParamCount(int paramCount)
         {
-            int inputParamCount = paramCount - 1; // since the return type is in the array
+            var inputParamCount = paramCount - 1; // since the return type is in the array
             switch (inputParamCount)
             {
                 case 0:
-                    return typeof(AWS.LambdaNoParam).FullName;
+                    return typeof(AWS.LambdaNoParamAsync).FullName;
                 case 1:
-                    return typeof(AWS.LambdaOneParam).FullName;
+                    return typeof(AWS.LambdaOneParamAsync).FullName;
                 case 2:
-                    return typeof(AWS.LambdaTwoParams).FullName;
+                    return typeof(AWS.LambdaTwoParamsAsync).FullName;
+                default:
+                    throw new ArgumentOutOfRangeException("AWS Lambda handler number of params can only be 0, 1 or 2.");
+            }
+        }
+
+        internal static string GetSyncIntegrationTypeFromParamCount(int paramCount)
+        {
+            var inputParamCount = paramCount - 1; // since the return type is in the array
+            switch (inputParamCount)
+            {
+                case 0:
+                    return typeof(AWS.LambdaNoParamSync).FullName;
+                case 1:
+                    return typeof(AWS.LambdaOneParamSync).FullName;
+                case 2:
+                    return typeof(AWS.LambdaTwoParamsSync).FullName;
                 default:
                     throw new ArgumentOutOfRangeException("AWS Lambda handler number of params can only be 0, 1 or 2.");
             }
         }
 
         internal static void Debug(string str)
         {
-            if (string.Equals(EnvironmentHelpers.GetEnvironmentVariable(LogLevelEnvName), "debug", StringComparison.OrdinalIgnoreCase))
+            if (EnvironmentHelpers.GetEnvironmentVariable(LogLevelEnvName)?.ToLower() == "debug")
```

Comment:
The [documented](https://docs.datadoghq.com/tracing/troubleshooting/#tracer-debug-logs) way to do this across all APM is `DD_TRACE_DEBUG=true`.

Reply:
yes we use DD_LOG_LEVEL across all other serverless product so I'd rather stay with that not to confuse our customer if that's OK

Reply:
Makes sense! Thanks.